internal/observability/graph.go

```diff
@@ -178,104 +178,97 @@ func (g *graph) createEdge(key []byte, req *transport.Request, direction string)
+	serverErrLatencies *metrics.Histogram
 }
 
 // newEdge constructs a new edge. Since Registries enforce metric uniqueness,
 // edges should be cached and re-used for each RPC.
-func newEdge(logger *zap.Logger, reg *pally.Registry, req *transport.Request, direction string) *edge {
-	labels := pally.Labels{
-		"source":           pally.ScrubLabelValue(req.Caller),
-		"dest":             pally.ScrubLabelValue(req.Service),
-		"procedure":        pally.ScrubLabelValue(req.Procedure),
-		"encoding":         pally.ScrubLabelValue(string(req.Encoding)),
-		"routing_key":      pally.ScrubLabelValue(req.RoutingKey),
-		"routing_delegate": pally.ScrubLabelValue(req.RoutingDelegate),
-		"direction":        pally.ScrubLabelValue(direction),
+func newEdge(logger *zap.Logger, meter *metrics.Scope, req *transport.Request, direction string) *edge {
+	tags := metrics.Tags{
+		"source":           req.Caller,
+		"dest":             req.Service,
+		"procedure":        req.Procedure,
+		"encoding":         string(req.Encoding),
+		"routing_key":      req.RoutingKey,
+		"routing_delegate": req.RoutingDelegate,
+		"direction":        direction,
 	}
-	calls, err := reg.NewCounter(pally.Opts{
-		Name:        "calls",
-		Help:        "Total number of RPCs.",
-		ConstLabels: labels,
+	calls, err := meter.Counter(metrics.Spec{
+		Name:      "calls",
+		Help:      "Total number of RPCs.",
+		ConstTags: tags,
 	})
 	if err != nil {
 		logger.Error("Failed to create calls counter.", zap.Error(err))
-		calls = pally.NewNopCounter()
 	}
-	successes, err := reg.NewCounter(pally.Opts{
-		Name:        "successes",
-		Help:        "Number of successful RPCs.",
-		ConstLabels: labels,
+	successes, err := meter.Counter(metrics.Spec{
+		Name:      "successes",
+		Help:      "Number of successful RPCs.",
+		ConstTags: tags,
 	})
 	if err != nil {
 		logger.Error("Failed to create successes counter.", zap.Error(err))
-		successes = pally.NewNopCounter()
 	}
-	callerFailures, err := reg.NewCounterVector(pally.Opts{
-		Name:           "caller_failures",
-		Help:           "Number of RPCs failed because of caller error.",
-		ConstLabels:    labels,
-		VariableLabels: []string{"error"},
+	callerFailures, err := meter.CounterVector(metrics.Spec{
+		Name:      "caller_failures",
+		Help:      "Number of RPCs failed because of caller error.",
+		ConstTags: tags,
+		VarTags:   []string{"error"},
 	})
 	if err != nil {
 		logger.Error("Failed to create caller failures vector.", zap.Error(err))
-		callerFailures = pally.NewNopCounterVector()
 	}
-	serverFailures, err := reg.NewCounterVector(pally.Opts{
-		Name:           "server_failures",
-		Help:           "Number of RPCs failed because of server error.",
-		ConstLabels:    labels,
-		VariableLabels: []string{"error"},
+	serverFailures, err := meter.CounterVector(metrics.Spec{
+		Name:      "server_failures",
+		Help:      "Number of RPCs failed because of server error.",
+		ConstTags: tags,
+		VarTags:   []string{"error"},
```

Comment:
since we'll be reusing the "error" string when using the vectors, we should put it into a const

Reply:
On the same note, we need to modify all the uses of the vectors to include the variable tag names

Before:
`vector.Get("undefined")`

Now:
`vector.Get("error", "undefined")`

Reply:
Done and done.